api/src/main/java/net/kyori/adventure/text/AbstractComponentBuilder.java

```diff
@@ -289,7 +289,13 @@ private void prepareChildren() {
   @Override
   @SuppressWarnings("unchecked")
   public @NotNull B mergeStyle(final @NotNull Component that, final @NotNull Set<Style.Merge> merges) {
-    this.styleBuilder().merge(requireNonNull(that, "component").style(), merges);
+    final Style thatStyle = requireNonNull(that, "that").style();
+    if (thatStyle.isEmpty() && merges.isEmpty()) return (B) this;
+    if (!this.hasStyle()) {
+      this.style = thatStyle;
```

Comment:
does this case also need to take into account `merges`?

Reply:
Yes you're right, good spot!